apis/v1alpha2/httproute_types.go

```diff
@@ -241,7 +241,7 @@ type HTTPRouteRule struct {
 	Filters []HTTPRouteFilter `json:"filters,omitempty"`
 
 	// BackendRefs defines the backend(s) where matching requests should be
-	// sent. If unspecified, the rule performs no forwarding. If unspecified and
+	// sent. If unspecified or invalid, the rule performs no forwarding; if
```

Comment:
Should we further clarify what invalid means?
Does a regular k8s Service with no endpoints count as invalid?

Reply:
Good point, wouldn't hurt to clarify that a Service with no endpoints should also not have traffic forwarded for it either. Maybe something like this:

> A BackendRef is considered invalid when it refers to a non-existent resource or a Service that does not have any endpoints.

Reply:
Would be nice to clarify too if you have multiple backendrefs what happens. From the discussion in the issue, you'd lad balance across some valid and some that return 503s?

Reply:
I personally think it makes more sense to leave the weighting as requested. IE if a user says that 80% of requests should go to an invalid backend, could we just make that 80% of requests 503? Although that's harsh, it may be better than the alternative of sending all traffic to an alpha endpoint, a failure that could potentially go unnoticed for a long time.

Reply:
I just feel we need to call out whatever behavior is decided. In my head, you'd set status for the invalid but skip the invalid services for serving. But that's ok if others don't agree. 🙃

Given this model, if you only had one invalid service defined for a route, you'd serve 503s, bit would you set a status on the route?

Reply:
> I personally think it makes more sense to leave the weighting as requested. IE if a user says that 80% of requests should go to an invalid backend, could we just make that 80% of requests 503? Although that's harsh, it may be better than the alternative of sending all traffic to an alpha endpoint, a failure that could potentially go unnoticed for a long time.

This was the consensus in the linked issue #638.

You could set an unresolved refs condition somewhere to report the inability to resolve an invalid service name.

Reply:
So, something like this?
```suggestion
	// sent. If unspecified or invalid (refers to a non-existent resource or a Service with no endpoints),
	// the rule performs no forwarding; if
```
With another bit added afterwards:
```go
	// 503 responses must be sent so that the overall weight is respected; if an invalid backend is requested
	// to have 80% of requests, then 80% of requests must get a 503 instead.
```